### PR TITLE
Arsh/eng 1730 validation set cli support

### PR DESCRIPTION
**Pull request by azahed98**
*Have you read the [Contributing Guidelines](https://github.com/togethercomputer/together/blob/main/CONTRIBUTING.md)?*
Yes

Issue # ENG-1730

## Describe your changes

This PR adds support of `validation_file` and `n_evals` arguments for fine-tuning. There is light input validation to ensure the submitted job is valid.

### CODE REVIEW

File: src/together/resources/finetune.py [hunk 1]
```diff
@@ -277,6 +285,8 @@ async def create(
             training_file (str): File-ID of a file uploaded to the Together API
             model (str): Name of the base model to run fine-tune job on
             n_epochs (int, optional): Number of epochs for fine-tuning. Defaults to 1.
+            validation file (str, optional): File-ID of a file uploaded to the Together API for validation.
```
Comment (mryab) [inline src/together/resources/finetune.py:288]:
```suggestion
            validation file (str, optional): File ID of a file uploaded to the Together API for validation.
```

File: src/together/resources/finetune.py [hunk 2]
```diff
@@ -48,6 +50,8 @@ def create(
...
```
Comment (mryab) [inline src/together/resources/finetune.py:53]:
```suggestion
            validation file (str, optional): File ID of a file uploaded to the Together API for validation.
```

**Review by mryab — APPROVED** [review]
